addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlUtils.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.paramminer;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpSender;
+
+/** A utility class for URL bruteforce related operations for the paramminer add-on. */
+public class UrlUtils {
+    private static Logger logger = LogManager.getLogger(UrlUtils.class);
+    private static HttpSender sender =
+            new HttpSender(
+                    Model.getSingleton().getOptionsParam().getConnectionParam(),
+                    true,
+                    HttpSender.MANUAL_REQUEST_INITIATOR);
```

Comment:
This should not be manual, worth registering a new initiator.

Reply:
Like? :)

Reply:
`PARAM_MINER_INITIATOR`?

Reply:
Okay but How do I add it to HttpSender ?

Reply:
Create a pull request adding the constant. (You'll not be able to use the constant here yet just the number.)

Reply:
I think it should be 16

Reply:
I'll make another PR as soon as this one is merged :)

Reply:
That value is already in use by OAST, you can create the PR now, it's in the zaproxy repo.

Reply:
Okay do I have to insert any comments? Like 
```
// ZAP: 2022/06/07 Address deprecation warnings with HttpSenderParos.
```

Reply:
Yes, it should have one. https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/#licensing-guidelines

Reply:
Done